File: mobile/app/api/interfaces/transaction.ts
import { IAccount } from './account';
import { ICategory } from './category';
interface ITransactionCreate {
  name: string;
  description?: string;
  amount: number;
  categoryId?: number;
  roomId: number;
}

export interface IIncomeTransaction extends ITransactionCreate {
  sender: string;
  destination: number;
}

export interface IExpendTransaction extends ITransactionCreate {
  sender: number;
  destination: string;
}

export interface IPersonTransaction extends ITransactionCreate {
  sender: number;
  destination: number;
}

export interface ITransaction {
  id: number;
  name: string;
  description: string;
  amount: number;
  transactionTypeId: number;
  category: ICategory;
  sender: IAccount;
  destination: IAccount;
  createDate: string;
}

File: WebAPI/FinAnalyzer.Domain/Entities/Transaction.cs
namespace FinAnalyzer.Domain.Entities
{
    public class Transaction : BaseEntity
    {
        public string Name { get; set; }

        public string? Description { get; set; }

        public decimal Amount { get; set; }

        public TransactionType TransactionType { get; set; }
        public int TransactionTypeId { get; set; }

        public Category? Category { get; set; }
        public int? CategoryId { get; set; }

        public Account Destination { get; set; }
        public int DestinationId { get; set; }

        public Account Sender { get; set; }
        public int SenderId { get; set; }
        
        public Room? Room { get; set; }
        public int? RoomId { get; set; }
    }
}